File: solace/cmd/play/Prompt.java
package solace.cmd.play;

import solace.net.*;

/**
 * Command that allows players to set custom prompt formats.
 * @author Ryan Sandor Richards
 */
public class Prompt extends PlayStateCommand {
  public Prompt(solace.game.Character ch) {
    super("prompt", ch);
  }

  public void run(Connection c, String []params) {
    if (params.length < 2) {
      c.sendln("Please enter a prompt format string to set.");
      return;
    }

    if (params.length > 2) {
      c.sendln(
        "Setting first paramter as prompt (hint: try quotes '' or \"\")");
    }

    character.setPrompt(params[1]);
  }
}

File: game/commands/character/prompt.js
'use strict';

/**
 * Command that allows players to set custom prompt formats.
 * @author Ryan Sandor Richards
 */
Commands.add('prompt', function (player, params) {
  if (player.isMobile()) {
    return player.sendln('Mobiles do not have prompts.');
  }

  var prompt;
  if (params.length < 2) {
    prompt = Config.get("game.default.prompt");
  } else {
    prompt = params[1];
  }

  if (params.length > 2) {
    player.sendln("Setting prompt as: " + prompt + ' (try using quotes)');
  } else {
    player.sendln("Setting prompt as: " + prompt);
  }
  player.getCharacter().setPrompt(prompt);
});
